src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/EventCounterTest.cs

```diff
@@ -16,91 +15,247 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     /// </summary>
     public class EventCounterTest
     {
+        public EventCounterTest()
+        {
+            ClearConnectionPools();
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void EventCounter_HardConnectionsCounters_Functional()
+        {
+            //create a non-pooled connection
+            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) {Pooling = false};
+
+            using var conn = new SqlConnection(stringBuilder.ToString());
+
+            //initially we have no open physical connections
+            Assert.Equal(0, SqlClientEventSourceProps.ActiveHardConnections);
+            Assert.Equal(0, SqlClientEventSourceProps.NonPooledConnections);
+
```

Comment:
Can you also add the check for 
`_hardConnectsCounter - _hardDisconnectsCounter == _activeHardConnectionsCounter` after we open and close non-pooled connection ?

Reply:
Done